reorg of topics for new Code Security section in Security (#26887)
DataDog/documentation: content/en/security/code_security/troubleshooting/_index.md

---
title: Troubleshooting
disable_toc: false
aliases:
   - /code_analysis/troubleshooting/
---

If you experience issues setting up or configuring Datadog Code Security, use this page to start troubleshooting. If you continue to have trouble, [contact Datadog Support][1].

> ## Static Analysis (SAST)

Reviewer: ```suggestion
## Static Code Analysis (SAST)
```

> For issues with the Datadog Static Analyzer, include the following information in a bug report to Support as well as your Customer Success Manager.

Reviewer: ```suggestion
For issues with the Datadog static analyzer, include the following information in a bug report to Datadog Support.
```

### Results are not being surfaced in the Datadog UI

**If you are running Code Security on a non-GitHub repository**, ensure that the first scan is ran on your default branch (for example, a branch name like
`master`, `main`, `prod`, or `production`). After you commit on your default branch, non-default branches are analyzed. You can always configure your default branch in-app under [Repository Settings][4].

If you are using Datadog's analyzer, [diff-aware scanning][6] is enabled by default. If you running the tool within your CI pipeline, make sure that `datadog-ci` runs **at the root** of the repository being analyzed.

## Software Composition Analysis

> For issues with Datadog Software Composition Analysis (SCA), include the following information in a bug report to Support as well as your Customer Success Manager.

Reviewer: ```suggestion
For issues with Datadog Software Composition Analysis (SCA), include the following information in a bug report to Datadog Support.
```

### Results are not being surfaced in the Datadog UI

> **If you are running Code Security on a non-GitHub repository**, ensure that the first scan is ran on your default branch (for example, a branch name like

Reviewer: ```suggestion
**If you are running static scanning on a non-GitHub repository**, ensure that the first scan is ran on your default branch (for example, a branch name like
```

The generated lock file is used by [`osv-scanner`][7] to extract dependencies and generate an SBOM.

## No vulnerabilities detected by Software Composition Analysis

There are a series of steps that must run successfully for vulnerability information to appear either in the [Service Catalog Security View][16] or in the [Vulnerability Explorer][12]. It is important to check each step when investigating this issue. 

> ### Confirm ASM is enabled

Reviewer: ```suggestion
### Confirming runtime detection is enabled
```

> You can use the metric `datadog.apm.appsec_host` to check if ASM is running.

Reviewer: ```suggestion
If you have enabled runtime vulnerability detection, you can use the metric `datadog.apm.appsec_host` to check if SCA is running.
```

Reviewer: @gorkavicente is this the right metric? we're combining troubleshooting guides
